sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala

```diff
@@ -57,173 +55,170 @@ abstract class CentralMomentAgg(child: Expression) extends ImperativeAggregate w
 
   override def dataType: DataType = DoubleType
 
-  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType)
+  override def inputTypes: Seq[AbstractDataType] = Seq(DoubleType)
+
+  protected val count = AttributeReference("count", DoubleType, nullable = false)()
```

Comment:
Shall we call it `n`? Otherwise, we have both `count` and `n` in the code. It is not clear that `n` means `newCount`. If we rename it, we can use `n` and `newN`.